### DISCUSSION TITLE
Refactor discussion for stitching/aligning code

### DISCUSSION THREAD

**Discussion by bethac07** [broadinstitute/pooled-cell-painting-image-processing#29] [Announcements]
Questions coming up as I refactor, put through individually so we can make threading possible

**Comment by bethac07** [comment]
Step structure: the new workflow we are thinking of doing is stitch-align-crop - do we want to make all steps mandatory, or support sub-combos (and if so, which?)

**Reply by ErinWeisbart** [reply]
My inclination is to say support ALL THE THINGS!
But assuming we don't need/want to always be *extra*, I could see the argument for making them all mandatory for now (until we stumble into the inevitable situation of needing to support a specific sub combo because Biology). 

I think the sub combo question is intertwined with what we do/don't save out - if we are supporting sub combos then we should support saving out the intermediates that would be the inputs.

I can think of a couple arguments against forcing align:
- if it's not necessary given the dataset, then we're forcing extra compute that doesn't need to happen (since I think we would probably always want to include alignment in the final pipeline also?)
- we haven't used it at all yet, so hard to know how often it might introduce artifacts and if it ever does, would be a bummer to force it

Seems like it would be easy to make crop optional since it's the last step? Wouldn't even need to introduce a "Run Crop Bool" and instead could be determined by the value passed as the dimensions e.g. if tile size = 0 then skip crop?

**Comment by bethac07** [comment]
File saving - do we want to always save all step outputs, or only final steps?

**Comment by bethac07** [comment]
10x downsamples - do we want to always make, or make configurable?

**Comment by bethac07** [comment]
File structuring - right now, we save each tiled image + channel separately, but we need not. How do we want files - multichannel, multicycle, or just the current structure?

**Reply by ErinWeisbart** [reply]
I'm currently agnostic.
